File: make_figures/gears.py
import pandas as pd
import altair as alt
import os 

# Figure <gears>
X = {}
for experiment in [f"1.4.2_{i}" for i in range(1, 7) if i!=4]:
    X[experiment] = pd.read_parquet(f"../experiments/{experiment}/outputs/evaluationPerPert.parquet")
X = pd.concat(X.values())
X.reset_index(inplace=True)

# Create the Altair chart
alt.data_transformers.disable_max_rows()
X["chart_x"] = X["regression_method"].astype(str) + "_using_" + X["eligible_regulators"].astype(str)
for metric in ["mae", "mse_top_20"]:
    chart = alt.Chart(X).mark_boxplot().encode(
        y=alt.Y(metric, title=metric),
        x=alt.X('chart_x', title='Method and gene-set for GO graph'),
        column=alt.Column('perturbation_dataset', title='Perturbation Dataset'),
        row=alt.Row('desired_heldout_fraction', title='Desired Heldout Fraction'),
        color=alt.Color('eligible_regulators', title='Eligible Regulators')
    ).properties(
        width=300,  # Adjust the width as needed
        height=300  # Adjust the height as needed
    )
    os.makedirs("plots", exist_ok=True)
    chart.save(f'plots/fig_gears_{metric}.html', embed_options={'renderer':'svg'})

File: make_figures/gears.R
library(ggplot2)
library(dplyr)
library(stringr)
library(arrow)
library(magrittr)
setwd("/home/ekernf01/Desktop/jhu/research/projects/perturbation_prediction/cell_type_knowledge_transfer/perturbation_benchmarking/make_figures/")

# Figure <gears>
collect_experiments = function(experiments){
  X <- list()
  for (experiment in experiments) {
    print(experiment)
    filepath <- paste0("../experiments/", experiment, "/outputs/evaluationPerPert.parquet")
    X[[experiment]] <- arrow::read_parquet(filepath)
    X[[experiment]]$color_by %<>% as.character
    X[[experiment]]$refers_to %<>% as.character
    X[[experiment]]$question %<>% as.character
  }
  X <- bind_rows(X)
  return(X)
}

X = collect_experiments(c(
  "1.4.2_1",
  "1.4.2_2",
  "1.4.2_3",
  # "1.4.2_4",
  "1.4.2_5",
  "1.4.2_6",
  "1.4.2_7"
  # "1.4.2_8"
)
)
X$regression_method %<>% gsub("0$", "", .)
the_usual_levels = unique(X$regression_method)
X$regression_method %<>% factor(levels = unique(c("empty", "dense", "median", "mean", "celloracle human", the_usual_levels)))

X %<>%
  group_by(regression_method, eligible_regulators, perturbation_dataset, desired_heldout_fraction, data_split_seed) %>%
  summarise(across(c("mae", "mse_top_20"), mean))
X$desired_heldout_fraction %<>% 
  multiply_by(100) %>%
  paste0("Held-out : ", ., "%")
X$data_split_seed %<>% as.character()
for(metric in c("mae", "mse_top_20")){
  ggplot(X) + 
    geom_line(aes_string(x = "regression_method", 
                        y = metric, 
                        group = "data_split_seed",
                        color = "data_split_seed")) + 
    labs(x='', 
         y = metric) +
    facet_grid(perturbation_dataset~desired_heldout_fraction, scales = "free_y") + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) 
  ggsave(paste0('plots/fig_gears_', metric, '.pdf'), width = 8, height = 3)
}


